Sources/Shared/Request+Album.swift

```diff
@@ -29,13 +29,71 @@ public typealias AlbumRequest = Request<Album>
 /// `Request` that returns an array of `Album` objects.
 public typealias AlbumListRequest = Request<[Album]>
 
+fileprivate enum Key {
+    static let name = "name"
+    static let description = "description"
+    static let password = "password"
+    static let privacy = "privacy"
+}
+
 public extension Request {
-    
     /// Returns a new request to fetch a specific album.
     ///
     /// - Parameter uri: The album's URI.
     /// - Returns: Returns a new `Request` for an individual album.
     static func albumRequest(for uri: String) -> Request {
         return Request(path: uri)
     }
+
+    /// Returns a new request for creating an album.
+    /// - Parameter userURI: The URI for the current user, and owner of the album being created.
+    /// - Parameter name: The name of the album being created.
+    /// - Parameter description: An optional description for the album.
+    /// - Parameter privacy: The privacy parameter as a String. If none is specified it defaults to "anyone".
+    /// - Parameter password: An optional password parameter, only required when privacy is set to "password".
+    static func createAlbumRequest(
+        userURI: String,
+        name: String,
+        description: String? = nil,
+        privacy: String? = VIMPrivacy_Public,
+        password: String? = nil
+    ) -> Request {
+        var parameters = [String: String]()
+        parameters[Key.name] = name
+        parameters[Key.description] = description
+
+        privacy.map { parameters[Key.privacy] = $0 }
+        password.map { parameters[Key.password] = $0 }
+
+        return Request(method: .POST, path: userURI + "/albums", parameters: parameters)
+    }
+
+    /// Returns a new request for updating an exising album.
+    /// - Parameter albumURI: The URI for the album that will be updated.
+    /// - Parameter name: The name of the album.
+    /// - Parameter description: An optional description for the album.
+    /// - Parameter privacy: The privacy parameter as a String.
+    /// - Parameter password: An optional password parameter, only required when the privacy is set to "password".
+    static func updateAlbumRequest(
+        albumURI: String,
+        name: String,
+        description: String? = nil,
+        privacy: String?,
+        password: String? = nil
+    ) -> Request {
+        var parameters = [String: String]()
+        parameters[Key.name] = name
+        parameters[Key.description] = description
+
+        privacy.map { parameters[Key.privacy] = $0 }
+        password.map { parameters[Key.password] = $0 }
+
+        return Request(method: .PATCH, path: albumURI, parameters: parameters)
+    }
+
+    /// Returns a new request to delete the album for the given URI.
+    /// - Parameter albumURI: The album's URI.
+    static func deleteAlbumRequest(for albumURI: String) -> Request {
```

Comment:
this and all of the methods above would definitely read a lot better if we could just pass the Album object directly

```swift
let request = Request.delete(album)
```

Reply:
I agree for the most part, and maybe we could discuss refactoring these after this phase is out? These are the problems though:
- It breaks with the convention we currently have for requests (even if it's weird and not necessarily a good one).
- Not all requestors (like the one for creation) have access to an `Album` object, though most would.
- If we're moving toward a world where our models are extracted from the networking layer we wouldn't want these requests to depend on a precise implementation of the album/showcase object. Maybe something like a protocol instead, but it feels like we're still a ways off from that.